### PR TITLE
feat: new error types for destinations(persistiq, pardot, one_signal, ometria, mssql)

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/pardot/networkHandler.js [hunk 1]
```diff
@@ -73,13 +49,15 @@ const pardotRespHandler = (destResponse, stageMsg, stage) => {
...
-    .build();
+  const destinationStatusCode = statusAndStats?.status || 400;
+  throw new NetworkError(
+    `Pardot: ${response.err} ${stageMsg}`,
+    destinationStatusCode,
+    {
+      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(destinationStatusCode)
+    },
+    response
+  );
```
Comment (utsabc) [inline src/v0/destinations/pardot/networkHandler.js:60]: We are not setting AuthErrCategory for this?
This is Oauth right?

Comment (sanpj2292) [reply src/v0/destinations/pardot/networkHandler.js:60]:
```
throw new ErrorBuilder()
    .setStatus(statusAndStats.status)
    .setDestinationResponse(response)
    .setMessage(`Pardot: ${response.err} ${stageMsg}`)
    .setAuthErrorCategory(getAuthErrCategory(errorCode))
    .setStatTags(statusAndStats.stats)
    .build();
```
We are sending `authErrorCategory` via a method right ?!

Comment (utsabc) [reply src/v0/destinations/pardot/networkHandler.js:60]: I didn't understand for NetworkError(), the last argument is `authErrorCategory`

Comment (mihir-4116) [reply src/v0/destinations/pardot/networkHandler.js]: can you check it now?

**Review by utsabc — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**